worker/storageprovisioner/caasworker.go

```diff
@@ -123,20 +123,19 @@ func (p *provisioner) loop() error {
 			}
 			for i, appID := range apps {
 				appLifeResult := appsLife[i]
-				if appLifeResult.Error != nil && params.IsCodeNotFound(appLifeResult.Error) {
+				if appLifeResult.Error != nil && params.IsCodeNotFound(appLifeResult.Error) || appLifeResult.Life == life.Dead {
```

Comment:
The caas firewaller uses the same pattern as here so that would need fixing also:
https://github.com/juju/juju/blob/5245dd17ed85e2a98cbbc31050772c06b9619865/worker/caasfirewaller/worker.go#L119-L129

Reply:
Right, the `caasfirewaller ` and `caasfirewallersidecar` are having similar issue, fixed.